Assets/Scripts/InputOutputButtonHandler.cs

```diff
@@ -0,0 +1,15 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using UnityEngine;
+
+public class InputOutputButtonHandler : MonoBehaviour
+{
+    public PuzzleGenerator puzzleGenerator;
+
+    public void InputToOutput()
+    {
+        puzzleGenerator = GetComponent<PuzzleGenerator>();

```

Comment:
This line is redundant and should be removed. puzzleGenerator is set through the editor, and is directly accessible at runtime (I've verified this change).

I checked in the editor, you feed this button a GameState object. Unity will extract the PuzzleGenerator component from that GameState automatically and place in in the reference on line 8 (You also in effect don't get access to the GameState if you were expecting it).